File: educational-core/testSrc/com/jetbrains/edu/coursecreator/CCTestsUtil.java
package com.jetbrains.edu.coursecreator;

import com.jetbrains.edu.learning.PlaceholderPainter;
import com.jetbrains.edu.learning.courseFormat.AnswerPlaceholder;
import com.jetbrains.edu.learning.courseFormat.TaskFile;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

public class CCTestsUtil {

  private CCTestsUtil() {
  }

  public static String getPlaceholderPresentation(AnswerPlaceholder placeholder) {
    return "offset=" + placeholder.getOffset() +
           " length=" + placeholder.getLength() +
           " possibleAnswer=" + placeholder.getPossibleAnswer() +
           " placeholderText=" + placeholder.getPlaceholderText();
  }

  public static void checkPainters(@NotNull AnswerPlaceholder placeholder) {
    final Set<AnswerPlaceholder> paintedPlaceholders = PlaceholderPainter.getPaintedPlaceholder();
    if (paintedPlaceholders.contains(placeholder)) return;
    for (AnswerPlaceholder paintedPlaceholder : paintedPlaceholders) {
      if (paintedPlaceholder.getOffset() == placeholder.getOffset() &&
          paintedPlaceholder.getLength() == placeholder.getLength()) {
        return;
      }
    }
    throw new AssertionError("No highlighter for placeholder: " + CCTestsUtil.getPlaceholderPresentation(placeholder));
  }

  public static void checkPainters(@NotNull TaskFile taskFile) {
    final Set<AnswerPlaceholder> paintedPlaceholders = PlaceholderPainter.getPaintedPlaceholder();

    for (AnswerPlaceholder answerPlaceholder : taskFile.getAnswerPlaceholders()) {
      if (!paintedPlaceholders.contains(answerPlaceholder)) {
        throw new AssertionError("No highlighter for placeholder: " + CCTestsUtil.getPlaceholderPresentation(answerPlaceholder));
      }
    }
  }
}

File: educational-core/testSrc/com/jetbrains/edu/coursecreator/CCTestsUtil.kt
package com.jetbrains.edu.coursecreator

import com.jetbrains.edu.learning.placeholder.PlaceholderPainter.getPaintedPlaceholder
import com.jetbrains.edu.learning.courseFormat.AnswerPlaceholder
import com.jetbrains.edu.learning.courseFormat.TaskFile

object CCTestsUtil {
  private fun getPlaceholderPresentation(placeholder: AnswerPlaceholder): String {
    return "offset=${placeholder.offset} " +
           "length=${placeholder.length} " +
           "possibleAnswer=${placeholder.possibleAnswer} " +
           "placeholderText=${placeholder.placeholderText}"
  }

  @JvmStatic
  fun checkPainters(placeholder: AnswerPlaceholder) {
    val paintedPlaceholders = getPaintedPlaceholder()
    if (paintedPlaceholders.contains(placeholder)) return
    for (paintedPlaceholder in paintedPlaceholders) {
      if (paintedPlaceholder.offset == placeholder.offset &&
          paintedPlaceholder.length == placeholder.length) {
        return
      }
    }
    throw AssertionError("No highlighter for placeholder: " + getPlaceholderPresentation(placeholder))
  }

  @JvmStatic
  fun checkPainters(taskFile: TaskFile) {
    val paintedPlaceholders = getPaintedPlaceholder()
    for (answerPlaceholder in taskFile.answerPlaceholders) {
      if (!paintedPlaceholders.contains(answerPlaceholder)) {
        throw AssertionError("No highlighter for placeholder: " + getPlaceholderPresentation(answerPlaceholder))
      }
    }
  }
}